core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -311,7 +311,7 @@ public static String formatWarning(final String s) {
      * @return the escaped string
      */
     public static String escape(String s) {
-        return s.replaceAll("(\\\\|\")", "\\\\$1");
+        return s.replaceAll("([\\\\\"])", "\\\\$1");
```

Comment:
I think the existing version is more readable and I doubt that the performance difference matters at all.